fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AvailableArrivalsPage.js

```diff
@@ -305,15 +305,23 @@ class CDUAvailableArrivalsPage {
                 selectedStarCell = selectedArrival.name;
                 selectedStarCellColor = mcdu.flightPlanManager.getCurrentFlightPlanIndex() === 1 ? "yellow" : "green";
             }
+
+            // we need to sort the transitions alphabetically for display, but keep track of their original index for the flightplan
+            const availableTransitions = [];
+            if (selectedApproach) {
+                availableTransitions.push(...selectedApproach.transitions.map((transition, index) => [transition, index]));
+                availableTransitions.sort(([a], [b]) => a.name.localeCompare(b.name));
+            }
```

Comment:
This is quite a big object assignment (potentially), maybe need to test the performance impact in QA

Reply:
It's creating an array with maybe 5-15 elements containing an array with 2 elements each, one being a reference to an object that already exists and the other a number. That should be pretty cheap. Ideally we wouldn't need to allocate at all, but in practice there isn't a good away around it, and this page loads nominally a couple of times per flight.